File: tests/fixtures/compare/sample.go
package main

import (
	"fmt"
	"os"
)

func main() {
	fmt.Println("hello")

	panic("fatal error")

	for i := 0; i < 10; i++ {
		defer fmt.Println(i)
	}

	os.Exit(1)
}

func empty() {}

func unreachable() int {
	return 42
	fmt.Println("never")
	return 0
}

File: tests/fixtures/compare/sample.java
// Realistic Java code with various issues
public class Sample {
    void printStuff() {
        System.out.println("hello");
        System.err.println("error");
    }

    void badCatch() {
        try {
            int x = 1 / 0;
        } catch (Exception e) {
        }
    }

    void stringCompare(String a, String b) {
        if ("hello" == a) {}
        if (a.equals(null)) {}
    }

    void empty() {}

    void nested() {
        try {
            try {
                int x = 1;
            } catch (Exception e) {
                e.printStackTrace();
            }
        } catch (Exception e) {}
    }

    void loopConcat() {
        String result = "";
        for (int i = 0; i < 10; i++) {
            result += "item" + i;
        }
    }
}
